libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp

```diff
@@ -276,6 +276,11 @@ bool AP_RCProtocol_CRSF::check_frame(uint32_t timestamp_us)
         return false;
     }
 
+    if (_frame.length + CRSF_HEADER_LEN < _frame_ofs) {
```

Comment:
So I think what you are testing is that you have a sufficiently long frame for a decode. I think it might be clearer if that was explicit so:
```
#define CRSF_FRAME_PAYLOAD_MIN 2
 if (_frame_ofs >= CRSF_HEADER_TYPE_LEN &&
        _frame.length < CRSF_FRAME_PAYLOAD_MIN) {
        return false;
    }
```
I stared at your logic for a long time trying to work out if it was correct or not! The problem with checking _frame_ofs is that I think there might be extra bytes after that we care about.

Reply:
> if (_frame_ofs >= CRSF_HEADER_TYPE_LEN &&
>         _frame.length < CRSF_FRAME_PAYLOAD_MIN) {
>         return false;
>     }

that would not fix it. The issue is _frame.length is too small. Your suggested change would make no difference at all.